src/buffer/out/textBuffer.cpp

```diff
@@ -166,6 +166,29 @@ ROW& TextBuffer::_getRowByOffsetDirect(size_t offset)
     return *reinterpret_cast<ROW*>(row);
 }
 
+// This function is "direct" because it trusts the caller to know that this row
+// could be the scratchpad row.
+size_t TextBuffer::_estimateOffsetOfLastCommittedRowDirect() const
+{
+    if (_commitWatermark == _buffer.get())
+    {
+        // Estimation failed. Act as though we have at least one non-scratchpad row committed.
+        // The caller will be passing this offset back to GetRowByOffset or similar later, which
+        // will commit it.
+        return 1;
+    }
+    const auto lastRowOffset = static_cast<ptrdiff_t>((_commitWatermark - 1) - _buffer.get()) / _bufferRowStride;
+    return gsl::narrow_cast<size_t>(lastRowOffset);

```

Comment:
i almost certainly blew up the x86 or audit builds and just haven't bothered to run locally. keep in mind.